api/app/src/main/kotlin/packit/service/BasePacketService.kt

```diff
@@ -52,7 +51,19 @@ class BasePacketService(
                     it.time.start, it.time.end
                 )
             }
+        val packetGroups = packets.groupBy { it.name }
+            .map { PacketGroup(it.key) }
+
         packetRepository.saveAll(packets)
+        saveUniquePacketGroups(packetGroups)
+    }
+
+    internal fun saveUniquePacketGroups(packetGroups: List<PacketGroup>)
+    {
+        val existingPacketGroupNames = packetGroupRepository.findAll().map { it.name }
+        val newPacketGroups =
+            packetGroups.filter { !existingPacketGroupNames.contains(it.name) }
+        packetGroupRepository.saveAll(newPacketGroups)
```

Comment:
I suspect pulling all packet groups out of the db to compare against a few new ones will become quite inefficient compared to doing a WHERE on the ones you're actually looking for.

Reply:
good catch yup implemented